### ISSUE TITLE
segmentation fault

### ISSUE THREAD

**Issue by mujtabahasan** [facebook/C3D#120]
I am trying to use your code for finetuning on a new dataset. For some video files, it just ends up giving segmentation fault. For some other videos, it gives following error:
I0512 20:52:25.651785 26466 solver.cpp:61] Solving DeepConv3DNet_ucf101_finetuning
F0512 20:52:32.099282 26466 math_functions.cpp:50] Check failed: status == CUBLAS_STATUS_SUCCESS (14 vs. 0)  CUBLAS_STATUS_INTERNAL_ERROR
*** Check failure stack trace: ***
    @     0x7f0f06d38daa  (unknown)
    @     0x7f0f06d38ce4  (unknown)
    @     0x7f0f06d386e6  (unknown)
    @     0x7f0f06d3b687  (unknown)
    @           0x4489a1  caffe::caffe_gpu_gemm<>()
    @           0x471a8f  caffe::InnerProductLayer<>::Backward_gpu()
    @           0x4378ea  caffe::Net<>::Backward()
    @           0x46d4ef  caffe::Solver<>::Solve()
    @           0x40abc1  main
    @     0x7f0f04816ec5  (unknown)
    @           0x40ce6e  (unknown)
    @              (nil)  (unknown)
Aborted (core dumped)

When I remove such videos, it trains fine. All the videos have same H264 codec. Does anyone have idea why it could be the case.

**Comment by dutran** [comment]
It seems like a CUDA error. Have you tried to finetune on UCF101? If so, does that get you to the same problem?

**Comment by mujtabahasan** [comment]
Hi. It is working fine with UCF101.. Is there any way of restarting the finetuning process from where it stopped if it breaks due to some reason like power cut.

**Comment by dutran** [comment]
1) It seem that you have the error at the fc layer (Backward). My bet is, you new dataset has a different number of classes (and it is bigger than 101), but you did not modify fc8 to adapt the change. Simple modify fc8 may fix your issue.

2) Yes, you can resume by train_net.bin solver_file solverstate_file

**Comment by mujtabahasan** [comment]
Yes, thanks a lot. That worked. In your paper, you have mentioned C3D with 3 nets. So, are the three nets different in architecture. How can I modify the one you have provided to obtain the three different architectures.
